### PR TITLE
Refactored the _user_mgt module

### PR DESCRIPTION
**Pull request by hiranya911**
* Refactored the data validators to return the values upon success. This pattern proved to be more readable and less verbose in the messaging module.
* Removed redundant helper functions, and in-lined the corresponding logic for clarity.
* Relaxed the exhaustive type checkers where appropriate, enabling more duck typed argument passing.

In addition to cleaning up the existing code, this also simplifies the implementation of the bulk user import API (being developed in a separate branch).

### CODE REVIEW

File: firebase_admin/_user_mgt.py [hunk 1]
```diff
@@ -37,66 +37,79 @@
...
                 'characters.'.format(uid))
+        return uid
 
     @classmethod
-    def validate_email(cls, email):
-        if not isinstance(email, six.string_types) or not email:
+    def validate_email(cls, email, required=False):
+        if email is None and not required:
+            return None
+        if not isinstance(email, six.string_types):
```
[the remark this answers is not in the document]
Comment (hiranya911) [reply firebase_admin/_user_mgt.py]: Actually an empty string check would be redundant in these cases. For example in case of emails we check:

```
parts = email.split('@')
if len(parts) != 2 or not parts[0] or not parts[1]:
  raise ValueError(...)
```

This fails for empty string. A similar check exists for phone numbers. We also have test coverage for empty email and phone number input.
